src/badge/style.tsx

```diff
@@ -0,0 +1,23 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { SYSTEM } from '../internal/environment';
+import { BadgeProps } from './interfaces';
+
+export function getBadgeStyles(style: BadgeProps['style']) {
+  if (!style?.root) {
+    return undefined;
+  }
+
+  return SYSTEM === 'core'
+    ? {
+        background: style.root.background,
+        borderColor: style.root.borderColor,
+        borderRadius: style.root.borderRadius,
+        borderStyle: style.root.borderWidth && 'solid',
```

Comment:
as this doesn't need to be dynamic (setting `border-style: solid` has no impact if border width is 0) I would suggest adding it via `styles.scss` instead.